moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/detail/ompl_constraints.h

```diff
@@ -334,4 +336,61 @@ std::shared_ptr<BaseConstraint> createOMPLConstraint(const moveit::core::RobotMo
                                                      const std::string& group,
                                                      const moveit_msgs::msg::Constraints& constraints);
 
-}  // namespace ompl_interface
\ No newline at end of file
+/******************************************
+ * Toolpath Constraint
+ * ****************************************/
+/** \brief A path constraint for the EEF useful for sanding, painting, drawing, welding, etc
+ * */
+class ToolPathConstraint : public BaseConstraint
+{
+public:
+  ToolPathConstraint(const moveit::core::RobotModelConstPtr& robot_model, const std::string& group,
+                     unsigned int num_dofs, const unsigned int num_cons = 6, double step = 0.01);
+
+  void parseConstraintMsg(const moveit_msgs::msg::Constraints& constraints) override;
+
+  void setPath(const moveit_msgs::msg::CartesianTrajectory& msg);
+
+  void function(const Eigen::Ref<const Eigen::VectorXd>& joint_values, Eigen::Ref<Eigen::VectorXd> out) const override;
+
+private:
+  // TODO: Why won't NearestNeighborsGNAT allow me to store pose as Eigen Transform
```

Comment:
@bostoncleek What kind of error do you get when you try to use an Eigen Transform as the template type?

Reply:
I was curious about this and it turns out the issue is there is no `==` operator implemented for these Eigen types:

```
/opt/ros/humble/include/ompl-1.6/ompl/datastructures/NearestNeighborsGNAT.h:551:102: error: no match for ‘operator==’ (operand types are ‘const Eigen::Transform<double, 3, 1>’ and ‘const Eigen::Transform<double, 3, 1>’)
  551 |                 if (dist < nbh.top().first || (dist < std::numeric_limits<double>::epsilon() && data == key))
```